### PR TITLE
Override node name

### PR DESCRIPTION
**Pull request by honnix**
Signed-off-by: Hongxin Liang <honnix@users.noreply.github.com>

# TL;DR
Override node name.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
More details are captured in #3092.

I tested this on our setup. Everything seems to work fine, except the `Nodes` tab of the execution detail view. While `Graph` and `Timeline` both show the overridden node name, `Nodes` tab still shows the original task name.

Also note that many changes were made by `black`.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3092

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/core/node.py [hunk 1]
```diff
@@ -108,6 +108,8 @@ def with_overrides(self, *args, **kwargs):
             )
         if "interruptible" in kwargs:
             self._metadata._interruptible = kwargs["interruptible"]
+        if kwargs.get("name") is not None:
```
Comment (honnix) [inline flytekit/core/node.py:111]: Real change here.

File: tests/flytekit/unit/core/test_node_creation.py [hunk 2]
```diff
@@ -401,3 +396,27 @@ def my_wf(a: str):
         _resources_models.ResourceEntry(_resources_models.ResourceName.CPU, "1"),
         _resources_models.ResourceEntry(_resources_models.ResourceName.MEMORY, "100"),
     ]
+
+
+@pytest.mark.parametrize(
```
Comment (honnix) [inline tests/flytekit/unit/core/test_node_creation.py:401]: Real change here.

File: tests/flytekit/unit/core/test_node_creation.py [hunk 3]
```diff
@@ -401,3 +397,24 @@ def my_wf(a: str):
         _resources_models.ResourceEntry(_resources_models.ResourceName.CPU, "1"),
         _resources_models.ResourceEntry(_resources_models.ResourceName.MEMORY, "100"),
     ]
+
+
+def test_name_override():
```
Comment (honnix) [inline tests/flytekit/unit/core/test_node_creation.py:402]: Real change here.

File: tests/flytekit/unit/core/test_node_creation.py [hunk 4]
```diff
@@ -87,6 +87,7 @@ def empty_wf2():
     wf_spec = get_serializable(OrderedDict(), serialization_settings, empty_wf2)
     assert wf_spec.template.nodes[0].upstream_node_ids[0] == "n1"
     assert wf_spec.template.nodes[0].id == "n0"
+    assert wf_spec.template.nodes[0].metadata.name == "t2"
```
Comment (honnix) [inline tests/flytekit/unit/core/test_node_creation.py:90]: Real change here.

File: flytekit/core/node.py [hunk 5]
```diff
@@ -108,6 +108,8 @@ def with_overrides(self, *args, **kwargs):
             )
         if "interruptible" in kwargs:
             self._metadata._interruptible = kwargs["interruptible"]
+        if "name" in kwargs:
```
Comment (honnix) [inline flytekit/core/node.py:111]: Real change here.

**Comment by honnix** [comment]
`FAILED tests/flytekit/unit/cli/pyflyte/test_run.py::test_pyflyte_run_cli - ValueError: I/O operation on closed file.` Not sure I understand what went wrong.

**Comment by eapolinario** [comment]
> `FAILED tests/flytekit/unit/cli/pyflyte/test_run.py::test_pyflyte_run_cli - ValueError: I/O operation on closed file.` Not sure I understand what went wrong.

It's not related to your change. I'm actively investigating this.

**Comment by honnix** [comment]
> > `FAILED tests/flytekit/unit/cli/pyflyte/test_run.py::test_pyflyte_run_cli - ValueError: I/O operation on closed file.` Not sure I understand what went wrong.
> 
> It's not related to your change. I'm actively investigating this.

@eapolinario Thanks for heads-up!

**Review by wild-endeavor — APPROVED** [review]

**Comment by honnix** [comment]
Hmm I guess I was using a different black version. I will try to fix that.

**Comment by wild-endeavor** [comment]
@honnix i just ran make fmt

**Review by wild-endeavor — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]

**Comment by honnix** [comment]
Sorry I missed the message. I reverted my commits.

**Comment by honnix** [comment]
> @honnix i just ran make fmt

Thanks. The pre-commit thing. Nice!

**Comment by honnix** [comment]
Thank you for merging this. Now I need your help to figure out why the UI is not consistent. I have attached some screenshots to help. Graph and Timeline are both good, but not Nodes. It feels like Nodes always takes the last part of the full name.

Screenshot 2022-11-22 at 22 26 29
Screenshot 2022-11-22 at 22 26 23
Screenshot 2022-11-22 at 22 26 08
